redbeat/schedulers.py

```diff
@@ -174,3 +188,36 @@ def schedule(self):
             d[entry.name] = entry
 
         return d
+
+    def tick(self, **kwargs):
+        if self.lock:
+            logger.debug('beat: Extending lock...')
+            rdb.pexpire(self.lock_key, int(self.lock_timeout * 1000))
+        return super(RedBeatScheduler, self).tick(**kwargs)
+
+    def close(self):
+        if self.lock:
```

Comment:
yes, that is the intent, clean shutdown releases the lock, termination would lead to waiting at most lock_timeout before the new beat dispatches tasks.